simplemma/rules/de.py

```diff
@@ -97,40 +108,25 @@ def apply_de(token: str, greedy: bool = False) -> Optional[str]:
     if len(token) < 7:
         return None
     # nouns
-    if token[0].isupper():  # and token.endswith("en"):
-        if token.endswith(("ereien", "heiten", "keiten", "ionen", "schaften", "täten")):
-            return token[:-2]
-        if token.endswith(("eusen", "icen", "logien")):
-            return token[:-1]
-        if token.endswith("ungen") and not (
-            "jungen" in token or "lungen" in token or "zungen" in token
-        ):
-            return token[:-2]
+    if token[0].isupper():
+        # noun endings/suffixes: regex search
+        match = NOUN_ENDINGS_DE.search(token)
+        if match:
+            # apply pattern
+            ending = next((g for g in match.groups() if g is not None), None)
+            if ending:
+                return token[: -len(ending)]
+            # lemma identified
+            return token
         # inclusive speech
-        # + Binnen-I: ArbeitnehmerInnenschutzgesetz?
+        # Binnen-I: ArbeitnehmerInnenschutzgesetz?
         if PLUR_ORTH_DE.search(token):
             return PLUR_ORTH_DE.sub(":innen", token)
-        # greedy search
-        if greedy:
-            # series of noun endings
-            match = NOUN_ENDINGS_DE.search(token)
-            if match and len(match[0]) > 2:
-                groups = [g for g in match.groups() if g is not None]
-                # lemma identified
-                if not groups:
-                    return token
-                # apply -en/-e/-n/-s patterns
-                return token[: -len(groups[0])]
-            # -end and gerunds
-            if GERUND_DE.search(token):
-                return ENDING_DE.sub("e", token)
     # mostly adjectives and verbs
     elif greedy and token[-1] in ENDING_CHARS_ADJ_DE:
```

Comment:
It would be ideal to get rid of this `greedy` here so rules are 100% independent of the greedy search.

An alternative could be to have 2 types of rules:
* `APPLY_RULES`
* `APPLY_GREEDY_RULES`
So the lemmatizer can try the normal rules and then the greedy rules if greedy is set to True.

Reply:
I agree, it would be best to simply get rid of the greedy option.